src/figuresClasses.ts

```diff
@@ -1,11 +1,108 @@
-export interface Figure {}
+type ColorType = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+enum ShapeEnum {
+  TRIANGLE = 'triangle',
+  CIRCLE = 'circle',
+  RECTANGLE = 'rectangle',
+}
+
+export interface Figure {
+  shape: ShapeEnum;
+  color: ColorType;
+  getArea: () => number;
+}
+
+export abstract class BaseFigure implements Figure {
+  shape: ShapeEnum;
+
+  color: ColorType;
+
+  constructor(shape: ShapeEnum, color: ColorType) {
+    this.color = color;
+    this.shape = shape;
+  }
+
+  abstract getArea(): number;
+}
+
+export class Triangle extends BaseFigure {
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(color: ColorType, a: number, b: number, c: number) {
+    super(ShapeEnum.TRIANGLE, color);
+    this.a = a;
+    this.b = b;
+    this.c = c;
```

Comment:
we can simplify it using the [Typescript Constructor Shorthand](https://dev.to/satansdeer/typescript-constructor-shorthand-3ibd)